modules/imports/src/main/scala/org/dhallj/imports/ResolveImportsVisitor.scala

```diff
@@ -184,27 +185,31 @@ final private class ResolveImportsVisitor[F[_] <: AnyRef](
 
     def importNonLocation(imp: ImportContext, mode: ImportMode, hash: Array[Byte]) =
       for {
-        _ <- if (parents.nonEmpty) ReferentialSanityCheck(parents.head, imp) else F.unit
+        _ <- ReferentialSanityCheck(parents.head, imp)
         _ <- rejectCyclicImports(imp, parents)
         r <- resolve(imp, mode, hash)
       } yield r
 
     for {
-      imp <- if (parents.isEmpty) canonicalize(i) else canonicalize(parents.head, i)
+      imp <- canonicalize(parents.head, i)
       result <- if (mode == ImportMode.LOCATION) importLocation(imp) else importNonLocation(imp, mode, hash)
     } yield result
   }
 }
 
 private object ResolveImportsVisitor {
-  def apply[F[_] <: AnyRef: Sync: Client]: F[ResolveImportsVisitor[F]] =
-    Sync[F].map2(ImportCache[F]("dhall"), ImportCache[F]("dhallj"))(apply[F](_, _))
+  def apply[F[_] <: AnyRef: Sync: Client](relativeTo: Path): F[ResolveImportsVisitor[F]] =
+    Sync[F].map2(ImportCache[F]("dhall"), ImportCache[F]("dhallj"))(apply[F](_, _, relativeTo))
 
   def apply[F[_] <: AnyRef: Sync: Client](semanticCache: ImportCache[F],
-                                          semiSemanticCache: ImportCache[F]): ResolveImportsVisitor[F] =
-    new ResolveImportsVisitor(semanticCache, semiSemanticCache, Nil)
-
-  def apply[F[_] <: AnyRef: Sync: Client](semanticCache: ImportCache[F]): ResolveImportsVisitor[F] =
-    new ResolveImportsVisitor(semanticCache, new ImportCache.NoopImportCache, Nil)
+                                          semiSemanticCache: ImportCache[F],
+                                          relativeTo: Path): ResolveImportsVisitor[F] =
+    //We add a placeholder filename "package.dhall" for the base directory as a Local import must have a filename
+    new ResolveImportsVisitor(semanticCache,
+                              semiSemanticCache,
+                              NonEmptyList.one(Local(relativeTo.resolve("package.dhall"))))
```

Comment:
I'll try it out tomorrow, but this will do the right thing if `relativeTo` is a file, right?

Reply:
Oh good question! The intention was definitely that it be a directory. I suspect it might not work. Will have a think about that one